Source/WebKit/Shared/UserInterfaceIdiom.h

```diff
@@ -27,12 +27,35 @@
 
 #if PLATFORM(IOS_FAMILY)
 
+#include <wtf/EnumTraits.h>
+
 namespace WebKit {
 
+enum class UserInterfaceIdiom : uint8_t {
+    Default,
+    SmallScreen,
+    Reality
+};
+
 bool currentUserInterfaceIdiomIsSmallScreen();
-void setCurrentUserInterfaceIdiomIsSmallScreen(bool);
+bool currentUserInterfaceIdiomIsReality();
+
+UserInterfaceIdiom currentUserInterfaceIdiom();
+void setCurrentUserInterfaceIdiom(UserInterfaceIdiom);
+
 bool updateCurrentUserInterfaceIdiom();
 
 }
 
+namespace WTF {
+template<> struct EnumTraits<WebKit::UserInterfaceIdiom> {
+    using values = EnumValues<
+        WebKit::UserInterfaceIdiom,
+        WebKit::UserInterfaceIdiom::Default,
+        WebKit::UserInterfaceIdiom::SmallScreen,
+        WebKit::UserInterfaceIdiom::Reality
+    >;
+};
+}
```

Comment:
Can we `serialization.in`?

Reply:
Maybe? I have to see what the caller is I guess.